### ISSUE TITLE
Suspend return value: 148

### ISSUE THREAD

**Issue by MedicineYeh** [caiogondim/bullet-train-oh-my-zsh-theme#177]
Hi,
I love this theme a lot, but there's a small flaw when I suspend process which is the return value would be 148 with "BULLETTRAIN_STATUS_ERROR_BG" color. 
I suggest adding a value detection in prompt_status() function. According to this [thread](http://unix.stackexchange.com/questions/62173/exit-status-of-148-upon-ctrlz) and [ABI](http://tldp.org/LDP/abs/html/exitcodes.html#EXITCODESREF), the value bigger than 128 is exit code 128 + signal __SIGTSTP__. Since the value has its meaning and text, we could show the text on the prompt instead of showing the return value.
For example, the value 148 should be "TSTP".
The text of return condition could be easily retrieved by using `kill -l $?`.

**Comment by dawikur** [comment]
Ohh, very good idea!

**Comment by caiogondim** [comment]
Ow
I was not aware of it
Bash and all it's hidden magic constants...

So, if `exit_code > 128` we can check it's value with `kill -l $?`?

**Comment by MedicineYeh** [comment]
Yes, you can simply try it out by opening a vim editor and then `ctrl+z` putting it to the background. After that, enter `kill -l $?`. You will see a different message. Magic~ :P
However, it might not be a good idea for values less than 128, since it would also use the signal number (__SIGTSTP__) to interpret the value, ex: a return value of 9 would be "KILL", which is not correct.

On more thing, don't use `kill -l $?` in `prompt_status()`, because the return value has been changed in [build_prompt()](https://github.com/caiogondim/bullet-train-oh-my-zsh-theme/blob/a70fadd45528490f1b484a7b9c8d416830357cd0/bullet-train.zsh-theme#L642).
Instead, you would want to use `kill -l $(($RETVAL - 128))`, which simply reuse the variable you have.

**Comment by caiogondim** [comment]
Could you make a PR?
You know lots more than me on this regard =)

**Comment by dbkaplun** [comment]
This functionality seems very interesting but the numeric exit code is most useful to me. Can we put this feature behind a flag, or have both the number and name output at once?

**Comment by MedicineYeh** [comment]
I ended up with the optional feature way. Please take a look at the PR [#179](https://github.com/caiogondim/bullet-train-oh-my-zsh-theme/pull/179).

**Comment by MedicineYeh** [comment]
By the way, in my own version, I specifically filter out the return value of 148, which is "suspend" (`ctrl;+z`).
It's just because I do this so often and I want my screen to be clean when there is nothing bad happen. 
This is just my habit, so I didn't put it in the PR. You can consider whether to put the following line in.
``` zsh
if [[ -n "$symbols" && $RETVAL -eq 148 ]]; then
  prompt_segment $BULLETTRAIN_STATUS_BG $BULLETTRAIN_STATUS_FG "⚙"
elif [[ -n "$symbols" && $RETVAL -ne 0 ]]; then
  ...
```
